code/TagField.php

```diff
@@ -17,37 +15,32 @@ class TagField extends DropdownField {
 	/**
 	 * @var bool
 	 */
-	protected $ajax = false;
+	protected $shouldLazyLoad = false;
 
 	/**
-	 * @var bool
-	 */
-	protected $readOnly = false;
-
-	/**
-	 * @var string
+	 * @var int
 	 */
-	protected $relationTitle = 'Title';
+	protected $lazyLoadItemLimit = 10;
 
 	/**
-	 * @var int
+	 * @var bool
 	 */
-	protected $ajaxItemLimit = 10;
+	protected $canCreate = true;
 
 	/**
-	 * @var null|string
+	 * @var string
 	 */
-	protected $recordClass;
+	protected $titleField = 'Title';
 
 	/**
 	 * @param string $name
-	 * @param null|string $title
-	 * @param array $source
-	 * @param array $value
-	 * @param bool $readOnly
+	 * @param string $title
+	 * @param null|DataList $source
+	 * @param null|DataList $value
+	 * @param bool $canCreate
 	 */
-	public function __construct($name, $title = null, $source = array(), $value = array(), $readOnly = false) {
-		$this->setReadOnly($readOnly);
+	public function __construct($name, $title = '', $source = null, $value = null, $canCreate = true) {
```

Comment:
Do we really need `$canCreate` as a constructor arg? The FormField API is confusing enough without varying the constructor args :)